docs/index.md

```diff
@@ -106,6 +109,35 @@ After installing `upctl`, you can run `upctl version` command to verify that the
 upctl version
 ```
 
+### Verify assets
+
+[GitHub artifact attestations](https://docs.github.com/en/actions/security-for-github-actions/using-artifact-attestations)
+and plain old checksum files are available for verifying release assets.
+
+=== "Attestations"
+
+    [Release asset artifact attestations](https://github.com/UpCloudLtd/upcloud-cli/attestations)
+    can be verified for example with the [GitHub CLI](https://github.com/cli/cli),
+    using the Linux x86_64 asset as an example:
+
+    ```sh
+    gh attestation verify \
+        /path/to/locally/downloaded/upcloud-cli_{{ latest_release }}_linux_x86_64.tar.gz \
+        --repo UpCloudLtd/upcloud-cli
+    ```
+
+    Attestations are available starting from version 3.16.0.
```

Comment:
Need to possibly adjust this, depending on which is the first release this becomes available with.